pkgs/development/libraries/quantlib/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, gitUpdater
+}:
+
+stdenv.mkDerivation rec {
+  pname = "quantlib";
+  version = "1.29";
+
+  src = fetchFromGitHub {
+    owner = "lballabio";
+    repo = "QuantLib";
+    rev = "QuantLib-v${version}";
+    sha256 = "sha256-TpVn3zPru/GtdNqDH45YdOkm7fkJzv/qay9SY3J6Jiw=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost ];
+
+  passthru.updateScript = gitUpdater { };
+
+  meta = with lib; {
+    description = "A free/open-source library for quantitative finance";
+    homepage = "https://quantlib.org";
+    platforms = platforms.unix;
+    license = licenses.quantlib;
```

Comment:
Not entirely sure about how licensing works, are we allowed to just use bsd3 here? Also the license page links to the bsd2 license but includes the text of bsd3 license for some reason